server/src/main/java/com/broadcom/lsp/cobol/service/CopybookService.java

```diff
@@ -14,14 +14,38 @@
  */
 package com.broadcom.lsp.cobol.service;
 
+import com.broadcom.lsp.cobol.core.model.CopybookModel;
 import com.broadcom.lsp.cobol.domain.event.api.EventObserver;
 import com.broadcom.lsp.cobol.domain.event.model.DataEvent;
 
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
 /**
- * Provide API definition to search for copybooks files and define the workspace folder opened in
- * the client when the LSP server triggered.
+ * Provide API definition to search for copybooks files.
+ * The service also caches copybook to reduce filesystem load.
  */
 public interface CopybookService extends EventObserver<DataEvent> {
```

Comment:
Sorry, not. We still need an observer because CopybookService waits for `AnalysisFinishedEvent` for sending downloading request to middleware.
I leave `EventObserver<DataEvent>` as is because in Databus refactoring I plan to get rid of EventObserver.

Reply:
Ok, makes sense